mpmp/utilities/analysis_utilities.py

```diff
@@ -433,48 +453,62 @@ def compare_control_ind(results_df,
                                 'delta_{}'.format(metric)])
 
 
-def compare_experiment(single_cancer_df,
-                       pancancer_df,
+def compare_experiment(condition_1_df,
+                       condition_2_df,
                        identifier='gene',
                        metric='auroc',
                        verbose=False):
 
     results = []
-    single_cancer_ids = np.unique(single_cancer_df[identifier].values)
-    pancancer_ids = np.unique(pancancer_df[identifier].values)
-    unique_identifiers = list(set(single_cancer_ids).intersection(pancancer_ids))
+    condition_1_ids = np.unique(condition_1_df[identifier].values)
+    condition_2_ids = np.unique(condition_2_df[identifier].values)
+    unique_identifiers = list(set(condition_1_ids).intersection(condition_2_ids))
 
     for id_str in unique_identifiers:
 
-        conditions = ((single_cancer_df[identifier] == id_str) &
-                      (single_cancer_df.data_type == 'test') &
-                      (single_cancer_df.signal == 'signal'))
-        single_cancer_results = single_cancer_df[conditions][metric].values
-
-        conditions = ((pancancer_df[identifier] == id_str) &
-                      (pancancer_df.data_type == 'test') &
-                      (pancancer_df.signal == 'signal'))
-        pancancer_results = pancancer_df[conditions][metric].values
-
-        if single_cancer_results.shape != pancancer_results.shape:
+        conditions = ((condition_1_df[identifier] == id_str) &
```

Comment:
For your statistical comparison when you compare model 1 (expression model) fold 1 with model 2 (27K methylation model) fold 1, does that mean that you have the same samples in fold 1? So you have the same seed you're using here?

By condition here do you mean model?

Reply:
> For your statistical comparison when you compare model 1 (expression model) fold 1 with model 2 (27K methylation model) fold 1, does that mean that you have the same samples in fold 1? So you have the same seed you're using here?

Yep, the folds have the same train/test samples across data types. This was something I made sure of as part of #48.

> By condition here do you mean model?

For all of my experiments I do mean model (i.e. comparing a model trained on data type 1 with a model trained on data type 2). I wanted to write this function in a general way because I can imagine using it for other comparisons in the future (for example, we could use it to compare single-cancer and pan-cancer models, like we were doing in https://github.com/greenelab/pancancer-evaluation), which is why I used "condition" instead of "model" - I feel like it's a more general term.